src/main/java/com/autosale/shop/service/impl/JwtTokenServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+package com.autosale.shop.service.impl;
+
+import com.autosale.shop.model.JwtTokensDTO;
+import com.autosale.shop.model.User;
+import com.autosale.shop.model.UserRole;
+import com.autosale.shop.service.JwtTokenService;
+import com.autosale.shop.service.UserService;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.UnsupportedJwtException;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class JwtTokenServiceImpl implements JwtTokenService {
+
+    private final String secretKey;
+
+    private final UserService userService;
+
+    private final int accessTokenTTL;
+
+    private final int refreshTokenTTL;
+
+    public JwtTokenServiceImpl(@Value("${jwt.secret}") String secret_key, UserService userService, @Value("${jwt.access_token_lifetime}") int access_token_lifetime, @Value("${jwt.refresh_token_lifetime}") int refresh_token_lifetime) {
```

Comment:
lifetime in minutes/seconds/mils/hours/days? How we can understand it from name?

Reply:
renamed